.github/workflows/marqo-migration.yml

```
@@ -0,0 +1,88 @@
+    - name: Build Migration Tool
+      run: |
+        cd internal/database/marqo/migration
+        go build -o marqo-migration .
+        ls -la # Debug: List files to confirm built binary
+
+    - name: Set Environment Variables
+      env: 
+        DEV_MARQO_API_BASE_URL: ${{ secrets.DEV_MARQO_API_BASE_URL }}
+        PROD_MARQO_API_BASE_URL: ${{ secrets.PROD_MARQO_API_BASE_URL }}
+        MARQO_API_KEY: ${{ secrets.MARQO_API_KEY }}      
+      run: |
```

actionlint (1.7.3)

shellcheck reported issue in this script: SC2086:info:2:74: Double quote to prevent globbing and word splitting

(shellcheck)